js/app/restaurant/menu.jsx

```diff
@@ -0,0 +1,272 @@
+  } else {
+    $('#menu_addSection').attr('disabled', true);
+    $('#add-menu-section').attr('disabled', true);
+    $form.find('[type="submit"]').attr('disabled', true);
+  }
+}
+
+function autoDismissMessages() {
+  setTimeout(function() {
+    $('#messages .alert').fadeOut();
+  }, 1000);
+}
+
+$(function() {
+
+  var $form = $('form[name="menu"]');
+
+  autoDismissMessages();
+
+  // Activate Bootstrap tooltips
+  $('[data-toggle="tooltip"]').tooltip();
+
+  // Show/hide inputs on page load
+  $form.find('.modifier-calculus-strategy').each((index, input) => renderCalculusStrategy($(input)));
+
+  $(document).on('click', '.close', function(e) {
+    e.preventDefault();
+    var selector = $(e.target).closest('.close').data('target');
+    $(selector).remove();
+  });
+
+  $(document).on('click', '[data-toggle="add-menu-item"]', function(e) {
+    e.preventDefault();
+    var selector = $(e.target).data('target');
+    var $target = $(selector);
+    addMenuItemForm($target);
+  });
+
+  $(document).on('click', '[data-toggle="add-menu-item-modifier"]', function(e) {
+    e.preventDefault();
+    var $target = $(e.target).prev();
+    addMenuItemModifierForm($target);
+    var $input = $target.closest('.menu-item-modifiers').find('.modifier-calculus-strategy');
+    renderCalculusStrategy($input);
+  });
+
+  $(document).on('click', '[data-toggle="add-menu-item-modifier-group"]', function(e) {
+    e.preventDefault();
+    e.preventDefault();
+    var selector = $(e.target).data('target');
+    var $target = $(selector);
+    addMenuItemModifierGroupForm($target);
+    var $input = $target.find('select.modifier-calculus-strategy');
+    renderCalculusStrategy($input);
+  });
+
+  $(document).on('click', '[data-toggle="remove-menu-section"]', function(e) {
+    e.preventDefault();
+
+    var selector = $(e.target).data('target');
+    var $target = $(selector);
+
+    var $copy = $target.detach();
+
+    var data = $form.serialize();
+    enableForm($form, false);
+
+    $.ajax({
+      url : $form.attr('action'),
+      type: $form.attr('method'),
+      data : data
+    })
+    .then(function(html) {
+      enableForm($form, true);
+      $copy.remove();
+      $('#messages').replaceWith(
+        $(html).find('#messages')
+      );
+      $('#menu_sections').replaceWith(
+        $(html).find('#menu_sections')
+      );
+      $('#add-section-wrapper').replaceWith(
+        $(html).find('#add-section-wrapper')
+      );
+      autoDismissMessages();
+    })
+    .catch(function(e) {
+      enableForm($form, true);
+    });
+  });
+
+  $('#menu_suggestions a').on('click', function(e) {
+    e.preventDefault();
+    $('#menu_addSection').val($(this).text());
+  });
+
+  $(document).on('show.bs.collapse', '[role="tabpanel"] .list-group-item .collapse', function () {
+    $(this).closest('.list-group-item').find('.show-description').remove();
+  });
+
+  const regex = /^menu\[sections\]\[[0-9]+\]\[items\]\[[0-9]+\]\[isAvailable\]$/
```

Comment:
This regexp doesn't work when the product is not saved yet (after clicking on 'Add element')